### PR TITLE
Add docs and tests for Traces class

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java [hunk 1]
```diff
@@ -94,22 +104,26 @@ public static TraceState startSpan(String operation) {
         }
     }
 
+    /**
+     * Subscribes the given span consumer to all "span completed" events. Subscribers are expected "cheap", i.e., do all
```
Comment (markelliot) [inline tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java]: are expected [to be] cheap

File: tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java [hunk 2]
```diff
@@ -94,22 +104,26 @@ public static TraceState startSpan(String operation) {
...
+    /** The inverse of {@link #subscribe}. */
     public static void unsubscribe(Subscriber subscriber) {
         SUBSCRIBERS.remove(subscriber);
     }
 
     /**
-     * Represents the event receiver for trace completion events. Implementations are invoked
-     * synchronously on the primary execution thread, and as a result should execute quickly.
+     * Represents the event receiver for trace completion events. Implementations are invoked synchronously on the
+     * primary execution thread, and as a result should execute quickly.
```
Comment (markelliot) [inline tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java]: should -> must

File: tracing/src/test/java/com/palantir/remoting1/tracing/TracesTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    public void testSubscribeUnsubscribe() throws Exception {
+        // no error when completing span without a registered subscriber
+        startAndCompleteSpan();
+
+        Traces.subscribe(subscriber1);
+        Traces.subscribe(subscriber2);
+        Span span = startAndCompleteSpan();
+        verify(subscriber1).consume(span);
+        verify(subscriber2).consume(span);
+        Mockito.verifyNoMoreInteractions(subscriber1, subscriber2);
```
Comment (markelliot) [inline tracing/src/test/java/com/palantir/remoting1/tracing/TracesTest.java]: static import the method to match the rest of the test

File: tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java (hunk 1 above)
Comment (uschi2000) [reply tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java]: done

File: tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java (hunk 2 above)
Comment (uschi2000) [reply tracing/src/main/java/com/palantir/remoting1/tracing/Traces.java]: done

File: tracing/src/test/java/com/palantir/remoting1/tracing/TracesTest.java (hunk 3 above)
Comment (uschi2000) [reply tracing/src/test/java/com/palantir/remoting1/tracing/TracesTest.java]: done

File: tracing/src/test/java/com/palantir/remoting1/tracing/TracesTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    public void testSubscribingIsIdempontent() throws Exception {
+        Traces.subscribe(subscriber1);
+        Traces.subscribe(subscriber1);
+        Span span = startAndCompleteSpan();
+        verify(subscriber1, times(1)).consume(span);
+        Mockito.verifyNoMoreInteractions(subscriber1, subscriber2);
+    }
+
+    @Test
+    public void testDoesNotNotifySubscribersWhenComplitingNonexistingSpan() throws Exception {
```
Comment (markelliot) [inline tracing/src/test/java/com/palantir/remoting1/tracing/TracesTest.java]: Compliting -> Completing

Comment (uschi2000) [reply tracing/src/test/java/com/palantir/remoting1/tracing/TracesTest.java]: done

**Review by markelliot — APPROVED** [review]

**Pull request merged**